### ISSUE TITLE
Request: Save colors with opacity channel (new save format)

### ISSUE THREAD

**Issue by S-Man42** [OpenRCT2/OpenRCT2#7319]
Hi,

I didn't find this one anywhere else. So I wanted to mention that. Concerning the new save format I had some thoughts about the object colors. In my opinion it would be nice to think about a four channel color save format - no matter if it would be used in first steps or not. With the opacity channel some future things could be much more simpler:
- transparent vehicles or tracks (or just poles e.g.) without messing around with the tile inspector
- semi transparent objects (you do not need any additional "glass" objects anymore)

I think it would be better to think about it before setting the specs of the format than making some creepy trys of inserting it later.

Just my 2cents :)

**Comment by duncanspumpkin** [comment]
That's not how rendering is done in rct2. You're basically asking for 32bpp which is a massive change to how everything is used.

**Comment by ipatix** [comment]
Just out of curiousity, is 32bpp something possible in the far future? I know OpenTTD does have support for 32bpp and I could imagine having higher color accuracy (for custom objects) and stuff like semi transparency (for edge anti aliasing) could improve looks quite a bit.

**Comment by IntelOrca** [comment]
@duncanspumpkin I think @S-Man42 is just saying that we should future proof our save format. Although it isn't that much of an issue as we will have a version-able and forwards compatible save format. The single byte for colour already allows for a transparency flag.

I think even in 32-bit colour I still prefer a discrete colour selection as opposed to a full range selector.

**Comment by S-Man42** [comment]
@IntelOrca Yes. I really wanted to mention it if one needs it in the future. Just one little byte which could make some future guys very happy and which isn't a bigger effort to implement perhaps. Just to think about it. To implement such a thing later could make much more effort than implement it at the beginning. But with a good versioning mechanism it wouldn't be a problem to implement it later on. :)

And of course: I am looking forward really luckily to see the planned invisible features :+1:
